### PR TITLE
Add PluralRules locale data shimming

### PR DESCRIPTION
**Pull request by bearfriend**
Support for Māori across browsers is inconsistent at best. Specifically for `Intl.PluralRules` (which is relied upon by `formatjs` to render `plural` arguments), no browsers have (proper) native support for Māori's plural categories. The `@formatjs/intl-pluralrules` polyfill also does not have data for Māori, and provides no value otherwise as `Intl.PluralRules` is well supported at this point.

We can rely on native `Intl.PluralRules` as long as we can shim locale data into it for `formatjs` to use. So, that's what we do here.

### CODE REVIEW

File: lib/PluralRules.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+			},
+		},
+		select(n, ord) {
+			return !ord && n === 1 ? 'one' : 'other';
+		}
+	}
+};
+
+function getCanonicalLocales(locales) {
+	const mappedLocales = [locales].flat().map(locale => {
```
Comment (bearfriend) [inline lib/PluralRules.js:23]: This is only needed for Safari on older OS versions (< ~macOS 12, based on sauce testing), which don't map the aliases back to `mi`.

File: package.json [hunk 2]
```diff
@@ -44,7 +44,6 @@
     "sinon": "^19.0.2"
   },
   "dependencies": {
-    "@formatjs/intl-pluralrules": "^1",
```
Comment (dlockhart) [inline package.json:47]: Just to clarify: the other things that this polyfill was providing (the plural rules locale data for browsers who don't support the native API) aren't needed at all anymore?

Comment (bearfriend) [reply package.json:47]: Ack! Safari 12 is the only browser that isn't blocked that doesn't support `Intl.Pluralules`. Do we want to keep it for that?

Comment (bearfriend) [reply package.json:47]: Discussed offline.

Brightspace loads... something in Safari 12 but it's completely unuseable so we're going ahead with removing the polyfill. We expect to block Safari 12 in the near future.

**Review by dlockhart — APPROVED** [review]

**Pull request merged**